lint/inverseMethodCall_checker.go

```diff
@@ -0,0 +1,57 @@
+package lint
+
+import (
+	"go/ast"
+	"go/types"
+	"strings"
+)
+
+func init() {
+	addChecker(&inverseMethodCallChecker{}, attrExperimental)
+}
+
+type inverseMethodCallChecker struct {
+	checkerBase
+}
+
+func (c *inverseMethodCallChecker) InitDocumentation(d *Documentation) {
+	d.Summary = "Detects inverse method execution"
+	d.Before = `type foo struct{}
+f := foo{}
+foo.bar(f)`
+	d.After = `type foo struct{}
+f := foo{}
+f.bar()`
+}
```

Comment:
`type foo struct{}` line seems redundant as it's clear that `foo` is a type (literal is used).